operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,304 @@
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for kueue
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// kueue configuration must not be changed once the object exists
+	// to change the configuration, one can delete the object and create a new object.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="values are immutable once set"
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config is the desired configuration
+	// for the kueue operator.
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+// +kubebuilder:validation:Enum=QueueNameOptional;QueueNameRequired
+type ManageJobsWithoutQueueNameOption string
+
+// ManageJobsWithoutQueueName allows to control what kind of
+// jobs kueue will manage.
+// Kueue jobs usually require kueue.x-k8s.io/queue-name on each job
+// to be opt-in for Kueue.
+const (
+	// QueueNameOptional means kueue will assume all workloads
+	// are to be gated by Kueue.
+	// This must be used with ManagedJobsNamespaceSelector.
+	QueueNameOptional ManageJobsWithoutQueueNameOption = "QueueNameOptional"
+	// QueueNameRequired means that the jobs require a queue label.
+	QueueNameRequired ManageJobsWithoutQueueNameOption = "QueueNameRequired"
+)
+
+// +kubebuilder:validation:Enum=Enabled;Disabled
+type EnabledOrDisabled string
+
+const (
+	Enabled  EnabledOrDisabled = "Enabled"
+	Disabled EnabledOrDisabled = "Disabled"
+)
+
+type KueueConfiguration struct {
+	// integrations are the workloads Kueue will manage
+	// kueue has integrations in the codebase and it also allows
+	// for external frameworks
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// expermentalFeatures are more expermental features
+	// that users can use to configure kueue.
+	// We do not guarantee that these features will yet be supported
+	// Once we are comfortable with these features, we will move this out of this list.
+	// +optional
+	ExpermentalFeatures *ExpermentalFeatures `json:"expermentalFeatures,omitempty"`
+}
+
+// These are more advanced features.
+type ExpermentalFeatures struct {
+	// resources provides additional configuration options for handling the resources.
+	// Supports https://github.com/kubernetes-sigs/kueue/blob/release-0.10/keps/2937-resource-transformer/README.md
+	// +optional
+	Resources Resources `json:"resources,omitempty"`
+	// featureGates are advanced gates that can control the feature gates that kueue sets
+	// in the configuration.
+	// +optional
+	FeatureGates map[string]EnabledOrDisabled `json:"featureGates,omitempty"`
```

Comment:
Can we drop all gating for now and make it opinionated within the operator?

Reply:
done.